extensions/knative/deployment/src/main/java/org/apache/camel/quarkus/component/knative/deployment/KnativeProcessor.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.knative.deployment;
+
+import java.util.List;
+
+import io.quarkus.arc.deployment.UnremovableBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.vertx.core.deployment.CoreVertxBuildItem;
+import io.quarkus.vertx.http.deployment.VertxWebRouterBuildItem;
+import org.apache.camel.component.knative.KnativeComponent;
+import org.apache.camel.component.knative.KnativeConstants;
+import org.apache.camel.component.knative.spi.KnativeEnvironment;
+import org.apache.camel.component.knative.spi.KnativeResource;
+import org.apache.camel.quarkus.component.knative.KnativeConsumerRecorder;
+import org.apache.camel.quarkus.component.knative.KnativeProducerRecorder;
+import org.apache.camel.quarkus.component.knative.KnativeRecorder;
+import org.apache.camel.quarkus.core.deployment.spi.CamelRuntimeBeanBuildItem;
+import org.apache.camel.quarkus.core.deployment.spi.CamelServiceFilter;
+import org.apache.camel.quarkus.core.deployment.spi.CamelServiceFilterBuildItem;
+import org.apache.camel.spi.ComponentCustomizer;
+
+class KnativeProcessor {
+
+    private static final String FEATURE = "camel-knative";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    List<UnremovableBeanBuildItem> unremovableBeans() {
+        return List.of(
+                UnremovableBeanBuildItem.beanTypes(KnativeEnvironment.class));
+    }
+
+    @BuildStep
+    List<ReflectiveClassBuildItem> reflectiveClasses() {
```

Comment:
Maybe we should open a Camel ticket to see if we can avoid the need for reflective access in future.

Reply:
Yeah, I think we can mark them with the `@Configurer` annotation and then we may get a reflection free bean binding

Reply:
mh, it looks like it is the case already https://github.com/apache/camel/blob/main/components/camel-knative/camel-knative-api/src/main/java/org/apache/camel/component/knative/spi/KnativeEnvironment.java#L41

Reply:
Okay

Reply:
that is not fixed yet

Reply:
https://issues.apache.org/jira/browse/CAMEL-18561